File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/activity/player/MySessionManagerListener.java
package tv.hdonlinetv.besttvchannels.movies.watchfree.activity.player;

import androidx.annotation.NonNull;

import com.google.android.gms.cast.CastStatusCodes;
import com.google.android.gms.cast.framework.CastSession;
import com.google.android.gms.cast.framework.SessionManagerListener;
import com.walhalla.ui.DLog;

public abstract class MySessionManagerListener implements SessionManagerListener<CastSession> {


    @Override
    public void onSessionStarting(@NonNull CastSession session) {
        DLog.d("@@@@@@" + session);
    }

    @Override
    public void onSessionResuming(@NonNull CastSession castSession, @NonNull String s) {
        DLog.d("@@@@@@" + castSession+" "+s);
    }

    @Override
    public void onSessionStartFailed(@NonNull CastSession castSession, int error) {
        String errorMessage = CastStatusCodes.getStatusCodeString(error);
        DLog.d("Session START failed. Error: " + errorMessage + " | Code: " + error);

    }

    @Override
    public void onSessionSuspended(@NonNull CastSession session, int reason) {
        DLog.d("@@@@@@" + session + " " + reason);
    }

    @Override
    public void onSessionResumeFailed(@NonNull CastSession session, int error) {
        String errorMessage = CastStatusCodes.getStatusCodeString(error);
        DLog.d("Session resume failed. Error: " + errorMessage + " | Code: " + error);

    }

    @Override
    public void onSessionEnding(@NonNull CastSession castSession) {
        DLog.d("@@@@@@" + castSession);
    }

    // Implement other methods as needed

}

File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/activity/player/MySessionManagerListener.kt
package tv.hdonlinetv.besttvchannels.movies.watchfree.activity.player

import com.google.android.gms.cast.CastStatusCodes
import com.google.android.gms.cast.framework.CastSession
import com.google.android.gms.cast.framework.SessionManagerListener
import com.walhalla.ui.DLog.d

abstract class MySessionManagerListener : SessionManagerListener<CastSession> {
    override fun onSessionStarting(session: CastSession) {
        d("@@@@@@$session")
    }

    override fun onSessionResuming(castSession: CastSession, s: String) {
        d("@@@@@@$castSession $s")
    }

    override fun onSessionStartFailed(castSession: CastSession, error: Int) {
        val errorMessage = CastStatusCodes.getStatusCodeString(error)
        d("Session START failed. Error: $errorMessage | Code: $error")
    }

    override fun onSessionSuspended(session: CastSession, reason: Int) {
        d("@@@@@@$session $reason")
    }

    override fun onSessionResumeFailed(session: CastSession, error: Int) {
        val errorMessage = CastStatusCodes.getStatusCodeString(error)
        d("Session resume failed. Error: $errorMessage | Code: $error")
    }

    override fun onSessionEnding(castSession: CastSession) {
        d("@@@@@@$castSession")
    } // Implement other methods as needed
}
